Packages/Tracking/Hands/Runtime/Scripts/HandModelManager.cs

```diff
@@ -0,0 +1,512 @@
+            {
+                handModelPairs.Add(new HandModelPair());
+            }
+
+            // Look for hand model pair in previously registered hands
+            for (int i = 0; i < handModelPairs.Count; i++)
+            {
+                HandModelBase potentialRegisteredHandPair;
+
+                if (handModel.Handedness == Chirality.Left)
+                {
+                    potentialRegisteredHandPair = handModelPairs[i].Right;
+                }
+                else
+                {
+                    potentialRegisteredHandPair = handModelPairs[i].Left;
+                }
+
+                if (handModel.gameObject.transform.parent == potentialRegisteredHandPair.gameObject.transform.parent)
+                {
+                    handModelPairIndex = i;
+                    return handModelPairIndex;
+                }
+            }
+
+            //If not found in previously registered hands, look for in the parent transform
+            HandModelBase potentialHandPair = handModel.gameObject.transform.parent.GetComponentsInChildren<HandModelBase>().FirstOrDefault(hmb => hmb != handModel);
+
+            handModelPairs.Add(new HandModelPair());
+
+            //Register the found pair, then return the pair index
+            if (potentialHandPair != null)
+            {
+                RegisterHandModel(potentialHandPair, handModelPairIndex);
+            }
+
+            return handModelPairIndex;
+        }
+
+        /// <summary>
+        /// Unregisters a hand with the Hand Model Manager.
+        /// If the HandModelPair no longer has a left and right hand it will be removed from the HandModelPair list
+        /// </summary>
+        /// <param name="handModel">The hand model to unregister</param>
+        public void UnregisterHandModel(HandModelBase handModel)
+        {
+            for (int i = 0; i < handModelPairs.Count; i++)
+            {
+                if (handModelPairs[i].ContainsHandModel(handModel))
+                {
+                    if (handModel.Handedness == Chirality.Left)
+                    {
+                        handModelPairs[i].Left = null;
+                    }
+                    else
+                    {
+                        handModelPairs[i].Right = null;
+                    }
+
+                    if (handModelPairs[i].Left == null && handModelPairs[i].Right == null)
+                    {
+                        handModelPairs.RemoveAt(i);
+                    }
+                    return;
+                }
+            }
+        }
+
+        #region Getters
+
+        /// <summary>
+        /// Returns all hands which will be active if a hand appears in the scene
+        /// This includes hands whose EnableDisable is not frozen, or are frozen in an active state
+        /// </summary>
+        /// <returns>Returns all hands which will be active if a hand appears in the scene</returns>
+        public List<HandModelBase> GetActiveHandsInScene()
```

Comment:
Is this not only if the hands have been registered?

Reply:
yeah... don't know how I got twisted here! Have renamed.